internal/properties/version.go

```diff
@@ -28,7 +28,14 @@ func compareSourceVersions(src api.SourceType, srcVer string, defVer string) err
 		srcVer, _, _ = strings.Cut(srcVer, "-")
 
 		// Major versions must match.
-		if semver.Compare(semver.Major("v"+srcVer), semver.Major("v"+defVer)) == 0 {
+		srcMajor := semver.Major("v" + srcVer)
+		defMajor := semver.Major("v" + defVer)
+		if semver.Compare(srcMajor, defMajor) == 0 {
+			return nil
+		}
+
+		// Use v8 definitions for v7.
+		if srcMajor == "v7" && defMajor == "v8" {
```

Comment:
Should probably put a FIXME there as this kind of logic should really belong in the source/target driver rather than in a generic internal function.

Reply:
Not sure what you mean? The source/target driver wouldn't know anything about how the properties handler interprets or validates versions. The driver supplies its version to the properties handler (`srcMajor`) which is validated against the named version in the properties yaml (`defMajor`). In this case we consider definitions with version `8.0` to be inclusive of `8.x.y` and `7.x.y`.

Reply:
What I'm saying is that v7 == v8 is a very VMware specific thing which therefore should belong in VMware specific logic, not in a generic helper that's also used by Incus and eventually other source/targets like OpenStack or Proxmox.

If any of those has a v7 and v8 version, we'd be in trouble.

Reply:
>  not in a generic helper that's also used by Incus and eventually other source/targets
This isn't that though. This function switches over `api.SourceType` and implements validation according to it, in this case specifically for `source: vmware` in the yaml file. If we ever have some other source type, it would never execute this code, and would need its own validator implemented, otherwise it would hit the error `return fmt.Errorf("Unsupported source %q", src)`

Reply:
As we add more sources, we'll probably move from in-place implementations over a switch statement to separate functions but the overall structure would be the same.

Reply:
Doh, I missed the `	case api.SOURCETYPE_VMWARE:` :)

Reply:
I suppose I can move the implementation to an exported helper in the `source/vmware.go` file, which is called here instead.

Reply:
Yeah, as we add more sources, we'll probably want to limit the amount of source driver logic in shared code down to a minimum, instead having the driver interface provide functions to access this kind of logic, keeping the drivers as self contained as possible.